### PR TITLE
Add config support for Hashicorp Vault plugin

### PR DESCRIPTION
**Pull request by linuxmail**
#### Pull Request (PR) description
This PR adds support for the the Vault plugin, which can be used as key storage provider.

#### This Pull Request (PR) fixes the following issues
Fixes: #428

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -222,8 +222,14 @@
   String $jvm_args                                    = $rundeck::params::jvm_args,
   Hash $kerberos_realms                               = $rundeck::params::kerberos_realms,
   String $key_password                                = $rundeck::params::key_password,
-  Enum['db', 'file'] $key_storage_type                = $rundeck::params::key_storage_type,
+  Enum['db', 'file', 'vault'] $key_storage_type       = $rundeck::params::key_storage_type,
   Stdlib::Absolutepath $keystore                      = $rundeck::params::keystore,
+  String $vault_keystorage_prefix                     = $rundeck::params::vault_keystorage_prefix,
```
Comment (bastelfreak) [inline manifests/init.pp]: For all those values, an empty string isn't valid, right? In that case, please use:
```
  String[1]
```

File: manifests/init.pp [hunk 2]
```diff
@@ -222,8 +222,14 @@
...
+  String $vault_keystorage_url                        = $rundeck::params::vault_keystorage_url,
```
Comment (bastelfreak) [inline manifests/init.pp]: `Stdlib::HTTPSUrl` should be a valid datatype

File: manifests/params.pp [hunk 3]
```diff
@@ -316,6 +316,13 @@
   $file_keystorage_keys = { }
   $file_keystorage_dir = "${framework_config['framework.var.dir']}/storage"
 
+  $vault_keystorage_prefix = 'rundeck'
+  $vault_keystorage_url = 'https://vault.example.com'
```
Comment (bastelfreak) [inline manifests/params.pp]: please default to undef here

File: manifests/params.pp [hunk 4]
```diff
@@ -316,6 +316,13 @@
...
+  $vault_keystorage_approle_approleid = '111111-222222-333333-44444-55555'
```
Comment (bastelfreak) [inline manifests/params.pp]: please default to undef here

File: manifests/params.pp [hunk 5]
```diff
@@ -316,6 +316,13 @@
...
+  $vault_keystorage_approle_secretid = 'aaaaaa-bbbbbbb-ccccccc-dddddd-eeeeee'
```
Comment (bastelfreak) [inline manifests/params.pp]: please default to undef here

File: manifests/config.pp [hunk 6]
```diff
@@ -22,6 +22,12 @@
   $file_default_mode            = $rundeck::file_default_mode
   $file_keystorage_dir          = $rundeck::file_keystorage_dir
   $file_keystorage_keys         = $rundeck::file_keystorage_keys
+  $vault_keystorage_prefix            = $rundeck::vault_keystorage_prefix
```
Comment (bastelfreak) [inline manifests/config.pp]: please align the `=`

**Comment by bastelfreak** [comment]
Hi @linuxmail, thanks for the PR! Can you please take a look at the inline comments I did and also check the used email address in the commits? It isn't  associated with your github account.

**Comment by linuxmail** [comment]
Hi,

I've added the fixes and looking later for other kind of stuff to do. Hopefully the mail address is also set on the Git repo :-)

**Comment by linuxmail** [comment]
I had to set the Vault entries to Optional, otherwise Travis wasn't happy ("..expected String, got undef..."

**Comment by bastelfreak** [comment]
Thanks for the PR!

**Pull request merged**